notes/dns-client-proxy.md

```diff
@@ -0,0 +1,667 @@
+return a BADPROXYPOLICY error.
+
+U = 1, UA = 0, A = 0
+
+: The proxy only tries only unencrypted transports.
+It is an error
+if either or both of the P or D flags is set and the proxy SHOULD return a
+BADPROXYPOLICY error.
+
+U = 0, UA = 1, A = 0
+
+: An effort is made to reach authenticated encryption, if that fails,
+unauthenticated encryption is tried. 
+It is an error
+if either or both of the P or D flags is set and the proxy SHOULD return a
+BADPROXYPOLICY error.
+
+U = 0, UA = 0, A = 1
+
+: The proxy only tries authenticated encryption. The P and D flags can be
+set to control which authentication mechanism has to be used.
+
+The P and D flags allow the application to require
+a specific authentication mechanism (PKIX or DANE). The meaning of the 
+flags is the following: 
+
+P = 0, D = 0
+
+: At least one of the two mechanisms has to validate for authenticated
+encryption to succeed.
+
+P = 1, D = 0
+
+: PKIX validation has to succeed, the status of DANE validation is ignored.
+
+P = 0, D = 1
+
+: A DANE record has to be present and be DNSSEC valid. 
+A DANE record has
+a Certificate Usage Field. For some values of this field (the values zero
+and one), DANE requires PKIX validation.
+In those cases, PKIX validation is also required according to the DANE
+specifications. For the values two and three, DANE does not require
+PKIX and because the P flag is zero, the result of PKIX validation has to
+be ignored.
+
+P = 1, D = 1
+
+: Both PKIX and DANE are required together. For PKIX, this means that
+PKIX validation has to succeed. For DANE it means that a DANE record 
+has to be present and be DNSSEC valid. Validation using the DANE record
+has to succeed. 
+
+Note that these two flags can only be used in combination with the A
+flag. The proxy SHOULD return a BADPROXYPOLICY error if either or both of the 
+P or D flags is set and the A flag is clear.
+
+The next flags provide more detailed control over which transports
+should be used or not. For each of 5 different transports
+(Do53, DoT, DoH with ALPN h2, DoH with ALPN h3, DoQ) there is a flag
+to allow (A53,AT,AH2,AH3,AQ) or disallow (D53,DT,DH2,DH3,DQ) the use of the
+transport. There is space to add more transports later. Note that setting
+the A flags and the D flag for a protocol (for example, setting both the
+A53 and the D53 flags) is not allowed and a proxy SHOULD reject such a request.
+
+To future proof applications, there is a single flag DD, that by default
+disallows transports that are not explicitly listed. With this flag clear,
+the application allows all transports that are not explicitly disallowed
+(including future transports). With the flag set, the
+application has to explicitly list which transports can be used.
+For example, by setting only DD and AT, the application forces the
+use of DoT.
+
+When DD = 0:
+
+* all transports are in the pool of potentially usable transports
+* D53, DT, DH2, DH3 and DQ remove those transports from the pool
+
```

Comment:
```suggestion
* The values of A53, AT, AH2, AH3 and AQ are ignored

```

Reply:
done

Reply:
Hmm, ignored is wrong, because earlier there is 'Note that setting
the A flag and the D flag for a protocol (for example, setting both the
A53 and the D53 flags) is not allowed and a proxy SHOULD reject such a request.'

I'll change ignored
 to irrelevant